### PR TITLE
Standardise instance count

### PR DESCRIPTION
**Pull request by cleherny**
## Goal
Standardise the number of instances between:
- the instances in `generate_tilesets.py`.
- the instances in COCO files.
- the instances in `assess_model.py`.

## How
- First, I chose to clip and explode the labels with the tiles before splitting the label parts into the different datasets in `generate_tilesets.py`. This gives the same distribution of instances as in the COCO files.  
- Second, to remain consistent, I also exploded the clipped labels with the tiles in `assess_model.py`. This seems to lower the recall. 

This is open to discussion!

### CODE REVIEW

File: scripts/generate_tilesets.py [hunk 1]
```diff
@@ -438,8 +438,13 @@ def main(cfg_file_path):
             logger.critical(e)
             sys.exit(1)
 
-        GT_tiles_gdf = gpd.sjoin(aoi_tiles_gdf, gt_labels_gdf, how='inner', predicate='intersects')
+        # split labels by tiles 
+        clipped_labels_gdf = misc.clip_labels(gt_labels_gdf, aoi_tiles_gdf, fact=0.9999)  
+        clipped_labels_gdf = clipped_labels_gdf.explode(ignore_index=True).drop(columns=['title','tile_id','x','y','z'])
```
Comment (GwenaelleSa) [inline scripts/generate_tilesets.py:443]: I don't have the "title" attribute in the project of the border points. It means it is not mandatory in the code.
In addition, if you have special columns, they are not dropped and duplicated with the suffixes "_left" and "_right" at line 446and then, the line 255 does not work anymore.
Wouldn't it be more versatile and give the same result to write: `clipped_labels_gdf = clipped_label_gdf.explode(ignore_index=True)[gt_labels_gdf.columns]`?

File: helpers/misc.py [hunk 2]
```diff
@@ -90,6 +90,7 @@ def clip_row(row, fact=fact):
 
     clipped_labels_gdf.drop(columns=['tile_geometry', 'index_right'], inplace=True)
     clipped_labels_gdf.rename(columns={'id': 'tile_id'}, inplace=True)
+    tiles_gdf.drop('tile_geometry', inplace=True, axis=1)
```
Comment (GwenaelleSa) [inline helpers/misc.py:93]: Unnecessary since tiles_gdf is not returned.

**Review by GwenaelleSa — COMMENTED** [review]
I tested your modification on the border points. I made two feedbacks so it could work.
Did you test the examples? Should I run the example of the roads?

My metrics dropped even though I currently don't have concav objects. It's weird and I will have to look further into it.

**Comment by GwenaelleSa** [comment]
I tested again and the variation in metrics is negligible. So it's ok for me if you modify line 443.

**Review by cleherny — COMMENTED** [review]
I have tested all the examples and it works. The metrics seem consistent with previous state.
Did you understood why your metrics were dropping for the border points?

**Review by GwenaelleSa — APPROVED** [review]
Ok :)

**Comment by GwenaelleSa** [comment]
I probably mixed up some files and it changed the metrics.

**Pull request merged**